test/utils/queueUtils.js

```diff
@@ -82,13 +93,44 @@ describe('test/utils/queueUtils.js - queue utils unit tests >', () => {
     return done();
   });
 
+  it('flush all buffered queue objects', (done) => {
```

Comment:
these tests mean my PR for the relatedLinks will have to wait.  We can postpone that PR until this one gets merged.